aten/src/ATen/native/cuda/Dropout.cu

```diff
@@ -110,10 +132,19 @@ fused_dropout_kernel_vec(at::cuda::detail::TensorInfo<const scalar_t, IndexType>
 
     // Perform the actual computation
     #pragma unroll
-    for (int ii = 0; ii < VEC; ii++) {
+    for (int ii = 0; ii < std::min(VEC, 4); ii++) {
```

Comment:
Won't this and the change below lead to incorrect results if non-ROCM devices/compilers start supporting vec length 8? What happens if we keep the original code unchanged?

Reply:
I agree with this comment better to keep all code in ROCm specific guards.

Reply:
I don't want to put all under ROCM, to avoid code duplication.
The code that touches the first dword is common for all cases (ROCM, NON-ROCM and VEC=2,4,8)
The code that touches the second dword is specific to ROCM and VEC=8.

if non-ROCM compilers need to support VEC=8 in future, all that is required is to remove the USE_ROCM checks